tests/load/pipeline/test_query_complexity_analyzer.py

```diff
@@ -0,0 +1,180 @@
+from typing import Any, Dict, List, Union
+import pytest
+
+from tests.load.utils import destinations_configs
+
+import sqlglot
+
+
+VARIOUS_QUERIES: List[Dict[str, Union[str, bool]]] = [
+    {"description": "star select", "query": "SELECT * FROM my_table", "complex": False},
```

Comment:
I would add the following tests:
* Select all columns but in a different order than they appear in the original table (should **not** be marked as complex)
* Select all columns but one column twice (should be marked as complex)
* Select in a way that all column names from the orignal column are there, but actually one of the columns is an alias of a different column: SELECT col1, col2, col2 as col3" should be marked as complex.